src/marqo/models/create_index_settings.py

```diff
@@ -0,0 +1,119 @@
+            storage_node_type (str, deprecated): Storage type for the index.
+            inference_node_count (int, deprecated): Number of inference nodes for the index.
+            storage_node_count (int, deprecated): Number of storage nodes for the index.
+            replicas_count (int, deprecated): Number of replicas for the index.
+            wait_for_readiness (bool): Marqo Cloud specific, whether to wait until the operation is completed or to proceed
+            without waiting for status; won't do anything if config.is_marqo_cloud=False.
+            inference_type (str): Inference type for the index.
+            storage_class (str): Storage class for the index.
+            number_of_inferences (int): Number of inferences for the index.
+            number_of_shards (int): Number of shards for the index.
+            number_of_replicas (int): Number of replicas for the index.
+
+        This class is used to manage default configuration settings for the Create Index function.
+        When creating an instance of this class, you can specify values for the configuration settings.
+        If a setting is not explicitly provided, its default value will be used.
+
+        Additionally, this class performs checks and raises deprecation warnings for deprecated parameters.
+        It also ensures that certain parameters, like `settings_dict`, are not specified along with other parameters.
+
+        Example:
+            >>> settings = CreateIndexSettings(model="my_model", normalize_embeddings=False)
+            >>> settings.model  # Accessing the model parameter
+            'my_model'
+            >>> settings.normalize_embeddings
+            False
+        """
+    treat_urls_and_pointers_as_images: bool = False
+    model: str = None
+    normalize_embeddings: bool = True
+    sentences_per_chunk: int = 2
+    sentence_overlap: int = 0
+    image_preprocessing_method: str = None
+    settings_dict: dict = None
+    inference_node_type: str = None
+    storage_node_type: str = None
+    inference_node_count: int = 1
+    storage_node_count: int = 1
+    replicas_count: int = 0
+    wait_for_readiness: bool = True
+    inference_type: str = None
+    storage_class: str = None
+    number_of_inferences: int = 1
+    number_of_shards: int = 1
+    number_of_replicas: int = 0
+
+    def __init__(self, **passed_parameters):
+        passed_parameters_without_none_values = {
+            key: value for key, value in passed_parameters.items() if value is not None
+        }
+        super().__init__(**passed_parameters_without_none_values)
+
+    @validator("*", pre=True)
+    def use_passed_parameters(cls, v, field):
+        return v
+
+    @root_validator(pre=True)
+    def handle_deprecated_parameters(cls, values):
+        deprecated_parameters_passed = [
+            deprecated_parameter for deprecated_parameter in index_defaults_deprecated_to_new_params_mapping.keys()
```

Comment:
you shouldn't need
`deprecated_parameter in index_defaults_deprecated_to_new_params_mapping.keys()`

This should be enough:
`deprecated_parameter in index_defaults_deprecated_to_new_params_mapping`